src/opcodes.c

```diff
@@ -20,6 +20,26 @@ int hexToDecimalMode(unsigned char hex) {
   return (hi * 10) + lo;
 }
 
+bool isAccumulatorAddressMode(unsigned int address) {
+  return address == 0;
+}
+
+void aslOpcode(unsigned int address, struct CPU *cpu, struct BUS *bus)
+{
+  unsigned char current = isAccumulatorAddressMode(address)
+    ? cpu->accumulator
+    : readBus(address, bus);
+  unsigned char result = current << 1;
+
+  cpu->status.carry = current & 0x80;
+  cpu->status.negative = result >> 7;
+  cpu->status.zero = result == 0x00;
+
+  return isAccumulatorAddressMode(address)
+    ? cpu->accumulator = result
+    : writeBus(address, result, bus);
```

Comment:
Alright, we do have something similar but this time it's write not only reads...
Please consider whether my previous comment is appropiated for this one too or not.

Reply:
:+1: